src/components/projects/projectId/playground/PlaygroundSearch.tsx

```diff
@@ -0,0 +1,202 @@
+import { Tooltip } from "@nextui-org/react";
+import { PlaygroundQuestion } from "@/src/types/components/projects/projectId/settings/playground";
+
+const PLAYGROUND_LIMIT_DEFAULT = 10;
+const PLAYGROUND_THRESHOLD_DEFAULT = -9999;
+
+export function PlaygroundSearch() {
+    const dispatch = useDispatch();
+
+    const projectId = useSelector(selectProjectId);
+    const onAttributeEmbeddings = useSelector(selectOnAttributeEmbeddings);
+    const attributes = useSelector(selectVisibleAttributesDataBrowser);
+
+    const [loading, setLoading] = useState(false);
+    const [selectedEmbedding, setSelectedEmbedding] = useState<Embedding>(null);
+    const [question, setQuestion] = useState("");
+    const [searchResults, setSearchResults] = useState(null);
+    const [limit, setLimit] = useState(PLAYGROUND_LIMIT_DEFAULT);
+    const [threshold, setThreshold] = useState(PLAYGROUND_THRESHOLD_DEFAULT);
+    const [metaDataFilter, setMetaDataFilter] = useState(null);
+
+    const [setCreationLoading, setSetCreationLoading] = useState(false);
+    const [createdSet, setCreatedSet] = useState(false);
+    const [questionHistory, setQuestionHistory] = useState<PlaygroundQuestion[]>();
+    const [showHistory, setShowHistory] = useState(false);
+    const [reformulationLoading, setReformulationLoading] = useState(false);
+
+    const getSearchResultsPost = useCallback(() => {
+        setLoading(true);
+        getSearchResults(projectId, selectedEmbedding?.id, question, limit, metaDataFilter, threshold, true, (result) => {
+            setLoading(false);
+            setCreatedSet(false);
+            setSearchResults(result);
+            refetchQuestionMemory();
+        });
+    }, [projectId, selectedEmbedding?.id, question, limit, metaDataFilter, threshold]);
+
+
+    const createSetFromRecords = useCallback(() => {
+        setSetCreationLoading(true);
+        if (createdSet) return;
+        setCreatedSet(true);
+        createEvaluationSet(projectId, question, searchResults.map((record) => record.id), (result) => {
+            setTimeout(() => {
+                setSetCreationLoading(false);
+            }, 1500);
+
+        });
+    }, [projectId, question, searchResults, createdSet]);
+
+    const handleHistoryClick = useCallback((entry: string) => {
+        setQuestion(entry);
+        setShowHistory(false);
+    }, []);
+
+    const handleReformulation = useCallback((apiKey) => {
+        setReformulationLoading(true);
+        getReformulationByQuestion(projectId, question, apiKey, (result) => {
+            if (!result?.reformulation) setQuestion("Error reformulating question.");
+            else setQuestion(result?.reformulation);
+            setReformulationLoading(false);
+        },
+            (error) => {
+                setReformulationLoading(false);
+                setQuestion("Error reformulating question.");
+            }
+        )
+    }, [projectId, question]);
```

Comment:
maybe also apikey

Reply:
i think no because apiKey is a parameter in the callback

Reply:
missed it in the callback, thanks